packages/core/src/electron-main/theia-electron-window.ts

```diff
@@ -115,8 +116,8 @@ export class TheiaElectronWindow {
 
     protected async handleStopRequest(onSafeCallback: () => unknown, reason: StopReason): Promise<boolean> {
         // Only confirm close to windows that have loaded our front end.
-        let currentUrl = this.window.webContents.getURL();
-        let frontendUri = this.globals.THEIA_FRONTEND_HTML_PATH;
+        let currentUrl = this.window.webContents.getURL(); // this URL string comes from electron, we can expect that this is properly encoded URL. For example, a space is `%20`
+        let frontendUri = FileUri.create(this.globals.THEIA_FRONTEND_HTML_PATH).toString(); // THEIA_FRONTEND_HTML_PATH is an FS path, we have to covert to an encoded URI string.
         // Since our resolved frontend HTML path might contain backward slashes on Windows, we normalize everything first.
         if (isWindows) {
```

Comment:
Suggestion: Since using `URI`s to encode everything normalizes paths anyway, we don't need to normalize the paths for Windows specifically. You can safely remove this block.